Sources/SwiftProtobuf/ProtobufJSONDecoding.swift

```diff
@@ -121,10 +121,12 @@ public struct ProtobufJSONDecoder {
     // TODO: ProtobufMessage here should be ProtobufFieldDecodableType to encompass both groups and messages
     // then this can merge with the decodeValue below...
     public mutating func decodeValue<M: ProtobufMessage>(key: String, message: inout M) throws {
+        guard let nameProviding = (M.self as? ProtoNameProviding.Type) else {
```

Comment:
I don't think so, since unlike the visitor pattern that takes a message in its initializer, the message being decoded into here isn't known until the method is called.